### PR TITLE
EZP-25880: Make ezuser.login case in-sensitive across databases, using case in-sensitive index

### PR DESCRIPTION
**Pull request by andrerom**
https://jira.ez.no/browse/EZP-25880

Redoes https://jira.ez.no/browse/EZP-19123 (#1561 & #1590) by using a case in-sensitive / lowercase index instead of new column. Objective is to not break db schema with legacy in 6.x.

- [x] Get tests to pass
- [x] Add update sql from 6.3, 6.1 and 5.4

### CODE REVIEW

**Comment by andrerom** [comment]
This is what I had in mind @emodric

**Comment by andrerom** [comment]
Review ping @alongosz @bdunogier @yannickroger @emodric

File: eZ/Publish/Core/Persistence/Legacy/User/Gateway/DoctrineDatabase.php [hunk 1]
```diff
@@ -55,9 +55,6 @@ public function createUser(User $user)
                 $this->handler->quoteColumn('login'),
                 $query->bindValue($user->login)
```
Comment (alongosz) [inline eZ/Publish/Core/Persistence/Legacy/User/Gateway/DoctrineDatabase.php]: Shouldn't we here - when creating a new user - also make `$user->login` lowercase (as it was done for `login_normalized` before)?

If I insert a new user e.g. 'Andrew' to a postgresql database, creating a case in-sensitive index will not make login match 'andrew'. Unless I'm missing something or it is handled elsewhere...

Comment (andrerom) [inline eZ/Publish/Core/Persistence/Legacy/User/Gateway/DoctrineDatabase.php]: > Shouldn't we here - when creating a new user - also make $user->login lowercase (as it was done for login_normalized before)?

no, we want to retain the login name user entered

> Unless I'm missing something or it is handled elsewhere...
see schema changes for postgres, and it should be more clear.

Comment (alongosz) [inline eZ/Publish/Core/Persistence/Legacy/User/Gateway/DoctrineDatabase.php]: Ok, I think I got it :)

**Comment by alongosz** [comment]
:+1:

**Comment by emodric** [comment]
Looks good to me! :+1: 

Will legacy have to be adopted to also load logins by lowercase as is done in e513d3732ee01cec454b124e4e102101af552c14?

**Comment by andrerom** [comment]
> Will legacy have to be adopted to also load logins by lowercase as is done in e513d37?

Hmm, that was the thing I was trying to avoid. But for postgress that seems to be the case yes.

**Comment by andrerom** [comment]
Thanks for the reviews :)

**Pull request merged**